app/scripts/modules/core/src/securityGroup/SecurityGroup.tsx

```diff
@@ -0,0 +1,148 @@
+  accountId: securityGroup.accountName,
+  name: securityGroup.name,
+  vpcId: securityGroup.vpcId,
+  provider: securityGroup.provider,
+});
+
+const ServerGroup = ({ serverGroup }: { serverGroup: IServerGroupUsage }) => (
+  <div className={`rollup-row ${serverGroup.isDisabled ? 'disabled' : ''}`} key={serverGroup.name}>
+    <UISref
+      to=".serverGroup"
+      params={{
+        region: serverGroup.region,
+        accountId: serverGroup.account,
+        serverGroup: serverGroup.name,
+        provider: serverGroup.cloudProvider,
+      }}
+    >
+      <a>
+        <CloudProviderLogo provider={serverGroup.cloudProvider} height="16px" width="16px" />
+        <span className="sp-padding-xs-left">{serverGroup.name}</span>
+      </a>
+    </UISref>
+  </div>
+);
+
+const ServerGroups = ({ serverGroups }: { serverGroups: IServerGroupUsage[] }) => {
+  // super confusing inverse-boolean here
+  if (!ReactInjector.$stateParams.hideServerGroups) {
+    return null;
+  }
+  return (
+    <div className="rollup-details-section col-md-6">
+      {serverGroups.length === 0 && <div className="small">No server groups</div>}
+      {serverGroups.length > 0 && (
+        <>
+          {serverGroups.map(serverGroup => (
+            <ServerGroup key={serverGroup.name} serverGroup={serverGroup} />
+          ))}
+        </>
+      )}
+    </div>
+  );
+};
+
+const LoadBalancer = ({
+  loadBalancer,
+  securityGroup,
+}: {
+  loadBalancer: ILoadBalancerUsage;
+  securityGroup: ISecurityGroup;
+}) => (
+  <div className="rollup-row" key={loadBalancer.name}>
+    <UISref
+      to=".loadBalancerDetails"
+      params={{
+        region: securityGroup.region,
+        accountId: securityGroup.account,
+        name: loadBalancer.name,
+        vpcId: securityGroup.vpcId,
+        provider: securityGroup.provider,
+      }}
+    >
+      <a>
+        <i className="fa icon-sitemap sp-padding-xs-right" /> {loadBalancer.name}
+      </a>
+    </UISref>
+  </div>
+);
+
+const LoadBalancers = ({ securityGroup }: { securityGroup: ISecurityGroup }) => {
+  // super confusing inverse-boolean here
+  if (!ReactInjector.$stateParams.hideLoadBalancers) {
+    return null;
+  }
+  return (
+    <div className="rollup-details-section col-md-6">
+      {securityGroup.usages.loadBalancers.length === 0 && <div className="small">No load balancers</div>}
+      {securityGroup.usages.loadBalancers.length > 0 && (
+        <>
+          {securityGroup.usages.loadBalancers.map(loadBalancer => (
+            <LoadBalancer key={loadBalancer.name} loadBalancer={loadBalancer} securityGroup={securityGroup} />
+          ))}
+        </>
+      )}
+    </div>
+  );
+};
+
+export const SecurityGroup = (props: ISecurityGroupProps) => {
+  const params = getSecurityGroupDetailsParams(props.securityGroup, props.application);
+  // don't use <UISrefActive> - it picks up load balancer and server groups details!
+  const active = ReactInjector.$state.includes('**.firewallDetails', params);
```

Comment:
FYI @christopherthielen this worked just fine without having to jump through any weird hoops, would recommend and use again.